### PR TITLE
[release/2.1] Add aspnet/Razor submodule for Microsoft.NET.Sdk.Razor

### PR DESCRIPTION
**Pull request by dagood**
Add https://github.com/aspnet/Razor to source-build to produce `Microsoft.NET.Sdk.Razor 2.1.2`.

For 2.2, after the auto-merge, I'll apply https://github.com/dotnet/source-build/compare/be27588b9c51c8c9bef9ef60f329eef41cf868f1...dagood:add-razor-sdk-2.2 to get the DLLs building. I decompiled the source-built DLLs and they're identical to the ones on the gallery. Doing a local build of those changes and this 2.1 PR to validate prebuilt baselines.

Once the aspnet repos are merged into aspnet/AspNetCore (not as submodules) these are the steps I'm thinking we'll need to do:
* Remove `src/aspnet-razor`, add `src/aspnetcore`.
* Rename project file to match.
  * Update base path.
  * Add handling of the global.json (present in AspNetCore but not Razor).
* Update patches for new paths.
  * (We actually don't have a mechanism to patch a submodule of a submodule, so we can't go ahead and use AspNetCore right now. The large number of submodules it would add before consolidation is a hindrance for some dev scenarios, too.)

FYI @natemcmaster 

Part of https://github.com/dotnet/source-build/issues/822

### CODE REVIEW

File: repos/aspnet-razor.proj [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <RazorSdkProjectDir>$(ProjectDirectory)src/Microsoft.NET.Sdk.Razor/</RazorSdkProjectDir>
+    <RazorSdkProjectFile>$(RazorSdkProjectDir)Microsoft.NET.Sdk.Razor.csproj</RazorSdkProjectFile>
+
+    <BuildNumber>30932</BuildNumber>
+
+    <BuildCommandArgs>pack</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) $(RazorSdkProjectFile)</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) /p:Configuration=$(Configuration)</BuildCommandArgs>
+    <!-- A nonfinal build adds "-t000" to the package version. (Prerelease builds are "final".) -->
+    <BuildCommandArgs>$(BuildCommandArgs) /p:IsFinalBuild=true</BuildCommandArgs>
```
Comment (crummel) [inline repos/aspnet-razor.proj]: Should this be equivalent to one of our `DropSuffix` or `Stabilized` properties?

Comment (dagood) [reply repos/aspnet-razor.proj]: My intent behind the comment is to explain why it shouldn't be. 😄 (Specifically, `UseStableVersions` in dependencies.props.) I'll make this clearer.

File: .gitmodules [hunk 2]
```diff
@@ -66,3 +66,6 @@
 [submodule "src/linker"]
 	path = src/linker
 	url = https://github.com/mono/linker.git
+[submodule "src/aspnet-razor"]
+	path = src/aspnet-razor
+	url = https://github.com/aspnet/Razor
```
Comment (natemcmaster) [inline .gitmodules]:
```suggestion
	url = https://github.com/aspnet/Razor
	branch = release/2.1
```

File: repos/aspnet-razor.proj (hunk 1 above)
Comment (dagood) [reply repos/aspnet-razor.proj]: Wait a sec... there's an issue for 2.2 where I ended up building `2.2.0-preview3-final` even though ProdCon built `2.2.0-preview3-35497`. I talked with @natemcmaster about it, who said `dotnet pack /p:BuildNumber=xyz /p:IsFinalBuild=true` (like it does now) should include the build number, but my 2.2 build just got to the point where I can see this still produces `2.2.0-preview3-final`.

Looking closer at version.props, it looks like `IsFinalBuild` should actually be `false`. Then, `PackageVersion` won't be set, and the SDK is free to combine `VersionPrefix` and `VersionSuffix`.

File: .gitmodules (hunk 2 above)
Comment (dagood) [reply .gitmodules]: AFAIK the only thing this allows is `git submodule update --remote`, and I don't think anyone uses that for source-build. From an infra perspective this is just another value to maintain and resolve conflicts for.

If someone made a case for adding these (to all the submodules that have applicable branches), I could get behind that, but for now I want to leave it out.

File: repos/aspnet-razor.proj (hunk 1 above)
Comment (natemcmaster) [reply repos/aspnet-razor.proj]: For 2.2 we stopped doing the `IsFinalBuild=true` step for previews, as you'll see on nuget.org: https://www.nuget.org/packages/Microsoft.NET.Sdk.Razor/2.2.0-preview3-35497.

But this PR is targeting 2.1, right? So, `IsFinalBuild=true` should be set for an RTM build of 2.1.x

**Comment by dagood** [comment]
@dotnet-bot test OSX10.12 Release
[Machine ran out of space.](https://ci.dot.net/job/dotnet_source-build/job/release_2.1/job/OSX10.12_Release_prtest/219/)

**Comment by dagood** [comment]
@dotnet-bot test CentOS7.1 Tarball Release
@dotnet-bot test RHEL7.2 Tarball Release
Straight-up timeouts.


I'm planning to merge regardless of source-build-CI if the Jenkins checks work. I'm working on getting the `source-build-CI` leg to re*try*, but that's blocked by an AzDO bug tracked at https://dnceng.visualstudio.com/internal/_workitems/edit/105. Re*build* is also blocked by a different AzDO bug. Only the `centos71 Offline` leg failed due to `Child node "2" exited prematurely` (somewhat known flakiness) so I'm not concerned.

(The MSBuild premature exit says it logged out some debug data, and I've merged https://github.com/dotnet/source-build/pull/836 to catch it next time. I'm skeptical that it actually wrote anything out, since it happened in Jenkins before and even though it's configured to save those files, it didn't find any.)

**Pull request merged**